crates/floresta-wire/src/p2p_wire/chain_selector.rs

```diff
@@ -325,8 +325,17 @@ where
                 // if they disagree on this current block, we need to look in the previous one
                 height -= 1;
             }
+            hash = self.chain.get_block_hash(height).unwrap();
         }
 
+        hash = self.chain.get_block_hash(fork).unwrap();
+
+        let (aggreed, _) = self
+            .grab_both_peers_version(peer1, peer2, hash, fork)
+            .await?;
+
+        hash = self.chain.get_block_hash(fork + 1).unwrap();
+
```

Comment:
This is one more issue I found in `find_who_is_lying` function, when we move forward in blockchain to find the fork, we also have to change the `BlockHash` along with the height.